backend/src/ee/services/dynamic-secret/providers/templateUtils.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable func-names */
+import handlebars from "handlebars";
+import RE2 from "re2";
+
+import { alphaNumericNanoId } from "@app/lib/nanoid";
+
+export const compileUsernameTemplate = ({
+  usernameTemplate,
+  randomUsername,
+  identityName,
+  unixTimestamp,
+  options
+}: {
+  usernameTemplate: string;
+  randomUsername: string;
+  identityName?: string;
+  unixTimestamp?: number;
+  options?: {
+    toUpperCase?: boolean;
+  };
+}): string => {
+  // Pre-process template to replace {{random-N}} patterns before compiling
+  let processedTemplate = usernameTemplate;
+  const randomPattern = /\{\{random-(\d+)\}\}/g;
```

Comment:
Why not just a normal handlebar partial of `random 12` instead of this. We don't need to then do this complex processing and keep it aligned with handle templating language